Sources/Recycling/GridViews/Ads/Cell/AdsGridViewCell.swift

```diff
@@ -250,28 +252,47 @@ public class AdsGridViewCell: UICollectionViewCell {
         }
 
         guard let viewModel = model, let dataSource = dataSource, viewModel.imagePath != nil else {
-            setImage(defaultImage)
+            setDefaultImage()
             return
         }
 
-        dataSource.adsGridViewCell(self, loadImageForModel: viewModel, imageWidth: frame.size.width) { [weak self] (fetchedModel, image) in
-            guard let model = self?.model else { return }
-            guard fetchedModel.imagePath == model.imagePath else { return }
-
-            let displayImage = image ?? self?.defaultImage
-            self?.setImage(displayImage)
+        if let cachedImage = dataSource.adsGridViewCell(self, cachedImageForModel: viewModel) {
+            setImage(cachedImage, animated: false)
+        } else {
+            dataSource.adsGridViewCell(self, loadImageForModel: viewModel, imageWidth: frame.size.width) { [weak self] (fetchedModel, image) in
+                guard let model = self?.model else { return }
+                guard fetchedModel.imagePath == model.imagePath else { return }
+
+                if let displayImage = image {
+                    self?.setImage(displayImage, animated: true)
+                } else {
+                    self?.setDefaultImage()
+                }
+            }
         }
     }
 
     // MARK: - Private
 
-    private func setImage(_ image: UIImage?) {
-        imageView.alpha = 0.0
+    private func setDefaultImage() {
+        imageView.image = defaultImage
+        self.imageView.alpha = 1.0
+    }
+
+    private func setImage(_ image: UIImage?, animated: Bool) {
         imageView.image = image
 
-        UIView.animate(withDuration: 0.2, delay: 0.0, options: [.curveEaseOut], animations: {
-            self.imageView.alpha = 1.0
-        })
+        let animations = { [weak self] in
+            self?.imageView.alpha = 1.0
+            self?.imageBackgroundView.backgroundColor = .clear
+        }
+
+        if animated {
+            imageView.alpha = 0.0
+            UIView.animate(withDuration: 0.2, delay: 0.0, options: [.curveEaseOut], animations: animations)
+        } else {
+            animations()
```

Comment:
If not `animated` then do `animations`... 🙃 

Maybe `animations` is not the best name for it? Not sure what is better though, but even `work` would probably be better

Reply:
Will fix!